### ISSUE TITLE
Mirror Termux packages for Android 5.x/6.x on IBiblio

### ISSUE THREAD

**Issue by ghost** [termux/termux-packages#4658]
> **UPDATE NOTE:** The last working Termux packaged for Android 5.x/6.x:
>
> - https://archive.org/download/termux-repositories-legacy/termux-repositories-legacy-24.12.2019.tar
> 
> **UPDATE2 NOTE:** Support for Android 5.x/6.x was *re-added to termux-app* (archived repo now hosted on official site), see:
> 
> - https://github.com/termux/termux-app/wiki/Termux-on-android-5-or-6

### Problem description

End of android-5/6 support on 2020-01-01

### Actual behavior

> From that date there will no longer be any updates to the package repositories' android-5 branches.
>
> An android-5 branch has also been created for termux-app, and the latest tagged version of the app (0.76) requires android 7.
>
> If you are using android 5 or android 6 on your device then the only way to continue to get updates after 2020-01-01 would be to upgrade to android 7 or newer (if possible).
> 
> - Q: What will happen to the android-5 branch? Will it be archived or completely removed?
>   - A: Branch will be available but packages may not.

According this note its not clear what would happen with packages repo for Android 5.x/6.x (removed, archived with read-only, or something else)

### Expected behavior

Mirror latest available *Termux* APKs & its packages for Android 5.x/6.x on IBiblio

- [ ] http://distro.ibiblio.org/termux/

## Additional information

Closes https://github.com/termux/termux-packages/issues/4467

### REFERENCES

- https://en.wikipedia.org/wiki/Ibiblio
- http://distro.ibiblio.org

**Comment by Symbian9** [comment]
Any news? Please, make it happen! (only 1 day left)

Backup all packages that, as for now, targeted to Android 5.x/6.x too (`termux/unstable-packages`, `termux/x11-packages`, etc.)

**Comment by sudomain** [comment]
Packages are available via the link in [this comment](https://github.com/termux/termux-packages/issues/4467#issuecomment-572008515). Note that the last working version of Termux for android 5/6 is Termux 0.75. This is available in Fdroid by enabling the 'F-Droid Archive' repo in settings.

If the archived packages are hosted elsewhere, I'm guessing that users will have to manually edit `$PREFIX/etc/apt/sources.list` to contain the url to the archived packages?

It's worth mentioning that as long as A5/6 users have a working version of proot, they can install [TermuxArch](https://github.com/SDRausty/TermuxArch) and recieve updated versions of packages like git, openssh, etc. from Arch Linux ARM. That'll be the main thing I use Termux for on my A5/6 devices.

**Comment by Symbian9** [comment]
> Packages are available

This not look like notmal repo. Instead it is just single large archive file.

I mean that mirror on iBiblio would be usable as normal repo. So, as for now, this issue/feature request not solved yet.

**Comment by sudomain** [comment]
> This not look like normal repo

I didn't mean to imply this, but the tar file contains .deb files which can be manually installed (though painfully) with `dpkg -i` .

Questions for devs: Are Termux repos just standard debian repos? Is there a tool on [this list](https://wiki.debian.org/DebianRepository/Setup) that is used for packaging?

**Comment by xeffyr** [comment]
> Are Termux repos just standard debian repos?

Yes, it is a standard Debian repository. Otherwise `apt` would just not work.

> Is there a tool on this list that is used for packaging?

We don't use utilities from this list.

Packages are created manually with script in https://github.com/termux/termux-packages/blob/master/scripts/build/termux_step_create_debfile.sh. Repository structure is constructed by https://bintray.com/.

**Comment by sudomain** [comment]
So ibiblio.org [hosts over 100 Linux distros and software projects on distro.ibiblio.org](https://www.ibiblio.org/software/), so it may be worthwhile. The application process seems (rather straight forward)[https://www.ibiblio.org/share/]. @xeffyr would it be ok if I applied for hosting of the Termux packages (for Android 5/6) on ibiblio on behalf of the Termux packages project?

**Comment by xeffyr** [comment]
Yes, it will be ok.

I should note that such mirroring is only useful for packages. Bootstrap environment is fetched from hardcoded URL which can't be changed without rebuilding application, so failure of `termux.net` will have serious effect anyway.

Reminding that `termux.net` is continuing to be available as there no plans to disable it.

**Comment by sudomain** [comment]
> Reminding that termux.net is continuing to be available as there no plans to disable it.

This is welcome news and pretty hilarious because I spent last night wrestling with `dpkg` to install packages from the archive, not realizing that `pkg install` still works. I can't speak for @Symbian9 , but for me this issue is resolved since termux.net still has the proot package for TermuxArch

**Comment by Symbian9** [comment]
Many packages which targeted to Android 5/6 akready removed from actual `termux.net` repo.

So, I still keep this issue as un-resolved.

**Comment by Zxce3** [comment]
any new news?

**Comment by spz2020** [comment]
`This issue/PR has been automatically marked as stale because it has not had recent activity. It will be closed if no further activity occurs. Thank you for your contributions.`
em.....

**Comment by Symbian9** [comment]
> This issue/PR has been automatically marked 

This issue still not solved. Keep it open till then.

**Comment by spectereye** [comment]
just met the same issue when using termux with android 5, maybe check https://github.com/termux/termux-app/wiki/Termux-on-android-5-or-6

**Comment by Zxce3** [comment]
> just met the same issue when using termux with android 5, maybe check https://github.com/termux/termux-app/wiki/Termux-on-android-5-or-6

nice info

**Comment by twaik** [comment]
Is it still relevant?

**Comment by twaik** [comment]
Closing this as irrelevant.
